test/pylib/scylla_cluster.py

```diff
@@ -493,7 +493,10 @@ async def stop_gracefully(self) -> None:
             self.is_running = False
             self.last_seed = None
 
-    async def add_server(self) -> str:
+    async def add_server(self, mark_dirty:bool=True) -> str:
+        """Add a new server to the cluster"""
+        if mark_dirty:
+            self.is_dirty = True
```

Comment:
I think you're missing the purpose of set_dirty: the idea is that every cluster that was manipulated with via rest api is "dirty", and every cluster that wasn't isn't, so it can be reused for the next test.

Reply:
Refactored as Nadav requested.